### ISSUE TITLE
Get supported checks from branch protection

### ISSUE THREAD

**Issue by ndench** [phstc/probot-merge-when-green#36]
As we discussed in #29, it would be good to get supported checks from branch protection and then add in any configured checks. I thought it better to create a new issue for this since we closed the other one out. Here are the details:

![image](https://user-images.githubusercontent.com/2062388/61765852-11aa3800-ae22-11e9-9fa2-415cbb2c7193.png)

I think we get a list of required checks using the [branch protection api](https://octokit.github.io/rest.js/#octokit-routes-repos-get-branch-protection) then merge it with a list of checks defined in the config. ie. if TravisCI is required and the user specifies `CircleCi` in the configuration, we must wait on both before merging the branch. This is because GitHub will block us from merging if only wait on CircleCi and TravisCi is not complete.

A couple of things to note:
- I think the branch protection API returns both checks and statuses together, so we likely need to implement #31 first
- We're using `app.owner.login` to identify checks at the moment [see here](https://github.com/phstc/probot-merge-when-green/blob/d9d21753ac9a3c0f83b5d67e921756811ee2b0df/src/mergeIfGreen.ts#L23). But it appears that the branch protection api will return the name of the check, so we'd have to update that.

**Comment by ndench** [comment]
As I mentioned in #41, I think it's a better idea to just check the [mergability of a branch](https://github.com/phstc/probot-merge-when-green/pull/41#discussion_r310322121) instead of getting a list of statuses required in branch protection. This will prevent us from trying to merge a branch that is blocked waiting on an approval for example. 

It also lets us build on top of the full power of branch protection for very little effort.

**Comment by phstc** [comment]
Just for the reference: https://github.com/phstc/probot-merge-when-green/pull/41#discussion_r310354437

**Comment by ndench** [comment]
I've just started getting into this, however it looks like the `mergeable` property only comes down through the [get a single pull request](https://developer.github.com/v3/pulls/#get-a-single-pull-request) API, and not through the [list pull requests associated with a commit](https://developer.github.com/v3/repos/commits/#list-pull-requests-associated-with-commit).

So I'll have to get rid of the interface I created in #41 and make the `statusSuccessHandler` do a request to get the pull request, as I mentioned in [this comment](https://github.com/phstc/probot-merge-when-green/pull/41/#discussion_r310095663).
